### PR TITLE
Allow max request header bytes to be 0

### PR DESCRIPTION
**Pull request by srijeet0406**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue 

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?

Go to TP and try setting the maxRequestHeaderBytes propert of a DS to 0 -> this should be allowed.
Also, while creating a new DS, the default value of this field should be 0.
Make sure the unit tests and API tests pass.

## If this is a bug fix, what versions of Traffic Control are affected?

- master

## The following criteria are ALL met by this PR

- [x] This PR does not include tests
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR does not include an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by rawlinp** [comment]
Should we bother updating the DB schema to make 0 the default there as well?

**Comment by srijeet0406** [comment]
@rawlinp Yeah I was thinking about that, wasn't sure if the outcome is worth having yet another migration. Personally, I try to keep the migrations to a minimum, but I dont mind adding one for the default value if you'd like. IMO the main functionality is to "allow" 0s, which we can do with this PR. But as I said, I'm fine either way.

**Comment by rawlinp** [comment]
Well, migrations are cheap, and it might be worth it for consistency's sake. It might be confusing to see the default as 0 in the API/UI but not in the DB.

File: traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html [hunk 1]
```diff
@@ -403,13 +403,13 @@ <h3>Current Value</h3>
                     <div class="form-group" ng-class="{'has-error': hasError(cacheConfig.maxRequestHeaderBytes), 'has-feedback': hasError(cacheConfig.maxRequestHeaderBytes)}">
                         <label class="has-tooltip control-label col-md-2 col-sm-2 col-xs-12" for="maxRequestHeaderBytes">Max Request Header Bytes<div class="helptooltip">
                                 <div class="helptext">
-                                    The maximum size (in bytes) of the request headers for this particular delivery service. Clearing this will result in the default (131072) being set for this DS.
+                                    The maximum size (in bytes) of the request headers for this particular delivery service. Clearing this will result in the default (0) being set for this DS.
```
Comment (mitchell852) [inline traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html:406]: You might want to tell the user what a value of 0 means exactly. Does it mean no max?

Comment (rawlinp) [reply traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html:406]: Good call, we might even want to reference the ATS parameter that this corresponds to: https://docs.trafficserver.apache.org/en/8.0.x/admin-guide/files/records.config.en.html#proxy-config-http-request-header-max-size.

**Comment by mitchell852** [comment]
image

If you set it to zero on the DS, does it then use the value found in `proxy.config.http.request_header_max_size`?

**Comment by srijeet0406** [comment]
> image
> 
> If you set it to zero on the DS, does it then use the value found in `proxy.config.http.request_header_max_size`?

Yes, you're right. I've made the doc change.

File: traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html [hunk 2]
```diff
@@ -403,13 +403,15 @@ <h3>Current Value</h3>
                     <div class="form-group" ng-class="{'has-error': hasError(cacheConfig.maxRequestHeaderBytes), 'has-feedback': hasError(cacheConfig.maxRequestHeaderBytes)}">
                         <label class="has-tooltip control-label col-md-2 col-sm-2 col-xs-12" for="maxRequestHeaderBytes">Max Request Header Bytes<div class="helptooltip">
                                 <div class="helptext">
-                                    The maximum size (in bytes) of the request headers for this particular delivery service. Clearing this will result in the default (131072) being set for this DS.
+                                    The maximum size (in bytes) of the request headers for this particular delivery service. Clearing this will result in the default <code>(0)</code> being set for this DS.
```
Comment (mitchell852) [inline traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html:406]: sorry to nit. How about this?

```The maximum size (in bytes) of the request headers for this particular delivery service. The default value is <code>0</code> which will limit the request header size to the value defined by the <a href="https://docs.trafficserver.apache.org/en/8.0.x/admin-guide/files/records.config.en.html#proxy-config-http-request-header-max-size" target="_blank">proxy.config.http.request_header_max_size</a> parameter.
```

Comment (srijeet0406) [reply traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html]: No worries, made that change.

**Comment by mitchell852** [comment]
ok, it looks like your previous migration  - `2020121800000000_add_max_request_header_size_delivery_service.sql` that actually created the `max_request_header_bytes` field set the value to `131072` (the default) for all ds's.  So my question is - do you need an update query in `2021012200000000_max_request_header_bytes_default_zero.sql` to set them all to zero? I don't know the answer, just asking the question.

**Comment by srijeet0406** [comment]
> ok, it looks like your previous migration - `2020121800000000_add_max_request_header_size_delivery_service.sql` that actually created the `max_request_header_bytes` field set the value to `131072` (the default) for all ds's. So my question is - do you need an update query in `2021012200000000_max_request_header_bytes_default_zero.sql` to set them all to zero? I don't know the answer, just asking the question.

Yes, you actually do need another migration to set the default to 0, for all new DS's.

**Review by mitchell852 — APPROVED** [review]
works as expected and resets all the values to 0 on migration. will merge once checks pass.

**Pull request merged**